lib/application/view-models/home/update_collection_details_vm.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:memo/application/pages/home/collections/update/update_collection_providers.dart';
+import 'package:memo/application/view-models/home/update_collection_vm.dart';
+import 'package:memo/application/widgets/theme/rich_text_field.dart';
+import 'package:memo/core/faults/exceptions/base_exception.dart';
+import 'package:memo/domain/validators/collection_validators.dart';
+
+final updateCollectionDetailsVM = StateNotifierProvider.autoDispose<UpdateCollectionDetailsVM, UpdatedDetailsState>(
+  (ref) => UpdateCollectionDetailsVMImpl(metadata: ref.read(updateDetailsMetadata)),
+  dependencies: [updateDetailsMetadata],
+  name: 'updateCollectionDetailsVM',
+);
+
+abstract class UpdateCollectionDetailsVM extends StateNotifier<UpdatedDetailsState> {
```

Comment:
Wasn't it possible to create a VM for each of them?

Reply:
I've made some tests creating an abstract VM that would be able to validate a single field from a generic type, this way we could easily instantiate a new VM for each field that we have. Would appreciate if you share your thoughts about it:


<details>
<summary>Field VM</summary>

```dart
abstract class FieldValidatorVM<T> extends StateNotifier<FieldValidatorState> {
  FieldValidatorVM({required this.validator, required FieldValidatorState state}) : super(state);

  final ValidationFunction validator;

  void validate(String updatedValue);
}

class FieldValidatorVMImpl<T> extends FieldValidatorVM<T> {
  FieldValidatorVMImpl({
    required ValidationFunction validator,
    required T value,
  }) : super(validator: validator, state: FieldValidatorState<T>(value: value));

  @override
  void validate(String updatedValue) {
    try {
      validator(updatedValue);
      state = state.copyWith(updatedValue: updatedValue);
    } on BaseException catch (exception) {
      state = state.copyForInvalid(exception);
    }
  }
}

class FieldValidatorState<T> extends Equatable {
  const FieldValidatorState({required this.value});

  final T value;

  FieldValidatorState copyWith({T? updatedValue}) => FieldValidatorState<T>(value: updatedValue ?? value);

  FieldValidatorInvalid copyForInvalid(BaseException exception) =>
      FieldValidatorInvalid<T>(value: value, exception: exception);

  @override
  List<Object?> get props => [value];
}

class FieldValidatorInvalid<T> extends FieldValidatorState<T> {
  const FieldValidatorInvalid({required T value, this.exception}) : super(value: value);

  final BaseException? exception;

  @override
  List<Object?> get props => super.props..add(exception);
}
```

</details>

And it could be used like:

<details>
<summary>Usage</summary>

```dart
final updateCollectionNameVM =
    StateNotifierProvider.family.autoDispose<FieldValidatorVM, FieldValidatorState<String>, ValidationFunction>(
  (ref, validator) => FieldValidatorVMImpl<String>(validator: validator, value: ref.read(updateDetailsMetadata).name),
  dependencies: [updateDetailsMetadata],
  name: 'updateCollectionNameVM',
);

final updateCollectionDescriptionVM = StateNotifierProvider.family
    .autoDispose<FieldValidatorVM, FieldValidatorState<RichTextEditingValue>, ValidationFunction>(
  (ref, validator) => FieldValidatorVMImpl<RichTextEditingValue>(
      validator: validator, value: ref.read(updateDetailsMetadata).description),
  dependencies: [updateDetailsMetadata],
  name: 'updateCollectionDescriptionVM',
);

final updateCollectionTagsVM =
    StateNotifierProvider.family.autoDispose<FieldValidatorVM, FieldValidatorState<List<String>>, ValidationFunction>(
  (ref, validator) =>
      FieldValidatorVMImpl<List<String>>(validator: validator, value: ref.read(updateDetailsMetadata).tags),
  dependencies: [updateDetailsMetadata],
  name: 'updateCollectionTagsVM',
);
```

</details>

But I thought that it was getting too complex to just integrate with these 3 fields, and I started to doubt from this solution. Besides that, I couldn't see any benefits on creating distinct but specific VMs for each field (like `UpdateCollectionNameVM`, `UpdateCollectionDescriptionVM` and `UpdateCollectionTagsVM`).

Reply:
> Besides that, I couldn't see any benefits on creating distinct but specific VMs for each field (like UpdateCollectionNameVM, UpdateCollectionDescriptionVM and UpdateCollectionTagsVM).

To not branch in a "off-topic" discussion, the TLDR of the benefits could be explained by referencing SOLID - and here, specifically the SRP (and somewhat the ISP) principles.

---

The solution is looking good, but I would say we would have to think a little bit more on it, more specifically about the "family" usage and it being a `StateNotifier`. I feel like we could improve it and make it significantly less verbose. Anyways, again, not work for this PR, we could start working on a draft for this one if you want.